pkg/gpu/nvidia/metrics/metrics.go

```diff
@@ -175,53 +171,58 @@ func (m *MetricServer) collectMetrics() {
 	}
 }
 
-func getGpuMetrics(device string, d *nvml.Device) (uint, uint64, error) {
-	status, err := gmc.collectStatus(d)
-	if err != nil {
-		glog.Errorf("Failed to get device status for %s: %v", device, err)
-		return 0, 0, err
+// getGpuMetricsInfo returns dutyCycle, usedMemory, totalMemory, uuid and deviceModel.
+func getGpuMetricsInfo(device string, d *nvml.Device) (uint, uint64, uint64, string, string, error) {
+	uuid, ret := d.GetUUID()
+	if ret != nvml.SUCCESS {
+		return 0, 0, 0, "", "", fmt.Errorf("failed to get GPU UUID: %v", nvml.ErrorString(ret))
 	}
-	mem := status.Memory
-	dutyCycle, err := gmc.collectDutyCycle(d.UUID, time.Second*10)
+	deviceModel, ret := d.GetName()
+	if ret != nvml.SUCCESS {
+		return 0, 0, 0, "", "", fmt.Errorf("failed to get GPU device model: %v", nvml.ErrorString(ret))
+	}
+
+	mem, ret := d.GetMemoryInfo()
+	if ret != nvml.SUCCESS {
+		return 0, 0, 0, "", "", fmt.Errorf("failed to get GPU memory: %v", nvml.ErrorString(ret))
+	}
+	dutyCycle, err := gmc.collectDutyCycle(uuid, time.Second*10)
 	if err != nil {
-		return 0, 0, fmt.Errorf("Failed to get dutyCycle: %v", err)
+		return 0, 0, 0, "", "", fmt.Errorf("failed to get dutyCycle: %v", err)
 	}
-	return dutyCycle, *mem.Global.Used, nil
+	return dutyCycle, mem.Used / (1024 * 1024), mem.Total / (1024 * 1024), uuid, deviceModel, nil
```

Comment:
nit: a const for `1024 * 1024` to make the unit conversion more clear can help

something like:

```go
const mib = 1024 * 1024
return dutyCycle, mem.Used / mib, mem.Total / mib, uuid, deviceModel, nil
```